app/_hub/kong-inc/ldap-auth/index.md

```diff
@@ -93,38 +104,49 @@ params:
       required: true
       default:
       value_in_examples: dc=example,dc=com
+      datatype: string
       description: Base DN as the starting point for the search; e.g., "dc=example,dc=com".
     - name: verify_ldap_host
       required: true
       default: "`false`"
+      datatype: boolean
       description: |
         Set to `true` to authenticate LDAP server. The server certificate will be verified according to the CA certificates specified by the `lua_ssl_trusted_certificate` directive.
     - name: attribute
       required: true
       default:
       value_in_examples: cn
+      datatype: string
       description: Attribute to be used to search the user; e.g., "cn".
     - name: cache_ttl
       required: true
       default: "`60`"
+      datatype: number
       description: Cache expiry time in seconds.
     - name: timeout
       required: false
       default: "`10000`"
+      datatype: number
       description: An optional timeout in milliseconds when waiting for connection with LDAP server.
     - name: keepalive
       required: false
       default: "`60000`"
-      description: An optional value in milliseconds that defines how long an idle connection to LDAP server will live before being closed.
+      datatype: number
+      description: |
+        An optional value in milliseconds that defines how long an idle connection to LDAP server will live before being closed.
     - name: anonymous
       required: false
       default:
+      datatype: string
       description: |
-        An optional string (consumer UUID) value to use as an "anonymous" consumer if authentication fails. If empty (default), the request will fail with an authentication failure `4xx`. The value must refer to the Consumer `id` attribute that is internal to Kong, **not** its `custom_id`.
+        An optional string (consumer UUID) value to use as an "anonymous" consumer if authentication fails. If empty (default), the request will fail with an authentication failure `4xx`.
+
+        **Note:** The value must refer to the Consumer `id` attribute that is internal to Kong, **not** its `custom_id`.
     - name: header_type
       required: false
```

Comment:
```suggestion
      required: true
```

Reply:
@murillopaula so that would be inconsistent with ldap auth advanced, which we just updated:

![image](https://user-images.githubusercontent.com/3756245/112483944-7defe400-8d47-11eb-8a34-4633dfc4a9fa.png)

The description says it's optional as well.

I know Aapo had gone through many of the plugins and changed booleans to required = true, but this schema is not explicit with required at this time.

   { header_type = { type = "string", default = "ldap" }, },

I don't have a ticket to make all configs required=true if they have a default.